apps/client/src/module/auth/login-form.tsx

```diff
@@ -15,12 +15,21 @@ import { SocialButton } from "../common/social-button";
 import { AuthFormContainer } from "./auth-form-container";
 import { login, LoginPayload } from "./auth-service";
 import { AuthRoute } from "./route";
+import {
+  EmailErrorMsgs,
+  emailPattern,
+  PasswordErrorMsgs,
+} from "@src/utils/validation-message";
 
 type LoginForm = z.infer<typeof formSchema>;
 // source of truth for this login form <= this is the law, is the king, is the absolute authority for this form - K总
 const formSchema = z.object({
-  email: z.string().email(),
-  password: z.string().min(8),
+  email: z.string()
+    .min(1, { message: EmailErrorMsgs.Required })
+    .email({ message: EmailErrorMsgs.Invalid })
+    .regex(emailPattern, { message: EmailErrorMsgs.Invalid }),
```

Comment:
What's special about this email pattern?

Reply:
The email pattern is left when joi was used before, has the same effect as.email(). I'll delete that line